Mix tags & text in aliases

dhague opened the pull request.

dhague committed:
Allow alias to have a mixture of plain text and @tags with typeahead suggestions for the final tag

dhague commented:
This allows the use of complex aliases such as:
`@hostname in @region`

rbak1 commented:
Thanks for the patch.  It's worth noting though that's there's two different aliasing mechanisms because of the inconsistent way that the api handles returning results and their dimensions. The typeahead fix was great, but the autoAlias fix only fixed one type.  If you want to copy new RegExp("@"+key, 'g') to the convertDataPoints method just below this I'd be happy to merge.  Or you can just wait and I've already pulled in your patch and fixed it up on the new grafana 3 branch I'm working on.

dhague committed:
Sort the keys of an object, longest first, before substituting aliases

This is so that when we are using them for search/replace, we don't accidentally replace a key which is a substring of another key.
Also: Apply global replace also to convertDataPoints()

dhague commented:
Good catch, thanks. I added `new RegExp("@"+key, 'g')` as suggested, and also fixed a bug - by replacing longer aliases first, we avoid an issue where "@short" could be replaced before "@shorthand", leaving a dangling "hand".
